prov/shm/src/smr_ep.c

```diff
@@ -179,16 +179,63 @@ static struct fi_ops_ep smr_ep_ops = {
 	.tx_size_left = fi_no_tx_size_left,
 };
 
-int smr_verify_peer(struct smr_ep *ep, int peer_id)
+void smr_generic_ctrl_format(struct smr_cmd *cmd, fi_addr_t id, uint32_t op)
 {
+	cmd->msg.hdr.op = op;
+	cmd->msg.hdr.op_flags = SMR_CTRL_MSG;
+	cmd->msg.hdr.addr = id;
+}
+
+static void smr_send_name(struct smr_ep *ep, fi_addr_t id)
+{
+	struct smr_region *peer_smr;
+	struct smr_cmd *cmd;
+	struct smr_inject_buf *tx_buf;
+
+	peer_smr = smr_peer_region(ep->region, id);
+
+	fastlock_acquire(&peer_smr->lock);
+
+	if (smr_peer_data(ep->region)[id].name_sent || !peer_smr->cmd_cnt)
+		goto out;
+
+	cmd = ofi_cirque_tail(smr_cmd_queue(peer_smr));
+	smr_generic_ctrl_format(cmd, id, ofi_ctrl_connreq);
+
+	tx_buf = smr_freestack_pop(smr_inject_pool(peer_smr));
+	cmd->msg.hdr.src_data = smr_get_offset(peer_smr, tx_buf);
+
+	cmd->msg.hdr.size = strlen(smr_name(ep->region)) + 1;
+	memcpy(tx_buf->data, smr_name(ep->region), cmd->msg.hdr.size);
```

Comment:
Is the name not stored in the shared memory region?

Reply:
It is, but this is to send the name of the sender if the target doesn't have it mapped yet so it needs the remote name in order to map it.